js_modules/dagster-ui/packages/ui-core/src/runs/QueuedRunCriteriaDialog.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Box,
+  Button,
+  Colors,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Icon,
+  Mono,
+  Table,
+  Tag,
+  Tooltip,
+} from '@dagster-io/ui-components';
+import * as React from 'react';
+import {Link} from 'react-router-dom';
+import * as yaml from 'yaml';
+
+import {RunTableRunFragment} from './types/RunTable.types';
+import {useRunQueueConfig} from '../instance/useRunQueueConfig';
+import {StructuredContentTable} from '../metadata/MetadataEntry';
+
+type TagConcurrencyLimit = {
+  key: string;
+  value?: string;
+  limit: number;
+};
+
+export const QueuedRunCriteriaDialog = ({
+  isOpen,
+  onClose,
+  run,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  run: RunTableRunFragment;
+}) => {
+  const runQueueConfig = useRunQueueConfig();
```

Comment:
As it's currently written, this query will be performed for every `QueuedRunCriteriaDialog`, even if they're never opened at all.

To avoid this, move the queryful behavior into the contents of the `Dialog`. The dialog contents will not be rendered until the dialog is opened.

```
export const QueuedRunCriteriaDialog = () => {
  return (
    <Dialog ...>
      <QueuedRunCriteriaDialogContents ... />
      <DialogFooter>
        ...
      </DialogFooter>
    </Dialog>
  );
};

const QueuedRunCriteriaDialogContents = () => {
  // Do query stuff
  // Render the table
};
```

You'll also probably want a loading state.